fling/utils/mp_utils.py

```diff
@@ -0,0 +1,41 @@
+import multiprocessing
+
+
+def client_trainer(client, kwargs):
+    # This is the function that each client will execute
+    # It will receive a task and its arguments, execute it, and return its result
+    return client.train(**kwargs)
+
+
+def client_tester(client, kwargs):
+    # This is the function that each client will execute
+    # It will receive a task and its arguments, execute it, and return its result
+    return client.test(**kwargs)
+
+
+def client_finetuner(client, kwargs):
+    # This is the function that each client will execute
+    # It will receive a task and its arguments, execute it, and return its result
+    return client.finetune(**kwargs)
+
+
+class MultiProcessLauncher:
+
+    def __init__(self, num_proc: int, task_name: str):
+        self.num_proc = num_proc
+        self.task_name = task_name
+
+    def launch(self, clients, **kwargs) -> list:
+        tasks = [(client, kwargs) for client in clients]
+        with multiprocessing.Pool(self.num_proc) as pool:
+            # Use starmap to apply the worker function to every task
+            # Each task is a tuple that contains the task object and the arguments
+            if self.task_name == 'train':
+                results = pool.starmap(client_trainer, tasks)
```

Comment:
After my tests and my research, I found that ’Multiprocess‘ uses ’fock‘ to create child processes that copy variables from the main process. Therefore, the ’tasks‘ you pass in here will be **deeply copied** by the main process. After the multi-process training is complete, the clients' model in the main process is not updated。
In this code framework, one solution is to return 'model.state_dict()' in the client's 'train' function and execute 'load_state_dict()' in the pipeline.  In my tests, it works.